addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHttpMethodScanRuleUnitTest.java

```diff
@@ -34,6 +43,78 @@ protected InsecureHttpMethodScanRule createScanner() {
         return new InsecureHttpMethodScanRule();
     }
 
+    @ParameterizedTest
+    @CsvSource({"PUT, json", "PUT, xml", "PATCH, json", "PATCH, xml"})
+    void shouldRaiseNoAlertsForPutOrPatchMethodsIfReturnJsonOrXml(String method, String contentType)
+            throws Exception {
+        // Given
+        String test = "/shouldRaiseNoAlertsForPutOrPatchMethodsIfReturnJsonOrXml/";
+        HttpMessage message = getHttpMessage(test);
+
+        nano.addHandler(
+                new NanoServerHandler(test) {
+
+                    @Override
+                    protected Response serve(IHTTPSession session) {
+                        Response response = newFixedLengthResponse("");
+
+                        if (session.getMethod().equals(Method.OPTIONS)) {
+                            response.addHeader("Allow", method);
+                            return response;
+                        }
+                        if (session.getMethod().equals(Method.PUT)
+                                || session.getMethod().equals(Method.PATCH)) {
+                            response.setMimeType("application/" + contentType);
+                            return response;
+                        }
+                        consumeBody(session);
+                        return response;
+                    }
+                });
```

Comment:
Better extract a class to avoid the duplication, pass the content-type as parameter.

Reply:
Can you please brief me about this? Like are you suggesting to do both the tests in a single test method? or is it something else?

Reply:
No he means create an internal class that extends NanoServerHandler with a constructor that accepts content-type then just use that in both test methods. (At least that’s my understanding)

Reply:
Actually this would require a some concepts of java which I am not completely familiar with (like constructor chaining in inheritance). I am still learning :)
So if this issue isn't that big, can we proceed without it?

Reply:
I can take care of it.

Reply:
Or link to existing examples, good way to learn :)

Reply:
e.g.
https://github.com/zaproxy/zap-extensions/blob/e95a2e525e66303482a29da3a7c43a73f535662c/addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRuleUnitTest.java#L149-L172

Reply:
Ohh okay, I understand now. Thanks for the example @thc202 

@kingthorin I will be able to do it now, thanks though :)


Will update it tomorrow as it's currently late night here in India. Have to attend the college too in morning :p

Reply:
Perfect, that had been my initial thought but I wasn’t in a good position to go looking for links 😉 Thanks @thc202